start_bot.py

```
@@ -0,0 +1,402 @@
+        json=room_layout,
+    )
+
+    if not response.ok:
+        print(f"could not create room layout: {room_layout['title']}")
+        response.raise_for_status()
+
+    return response.json()["id"]
+
+
+def create_room(room_layout):
+    response = requests.post(
+        f"{SLURK_API}/rooms",
+        headers={"Authorization": f"Bearer {API_TOKEN}"},
+        json={"layout_id": room_layout},
+    )
+
+    if not response.ok:
+        print(f"could not create room: {room_layout['title']}")
+        response.raise_for_status()
+
+    return response.json()["id"]
+
+
+def create_permissions(permissions):
+    response = requests.post(
+        f"{SLURK_API}/permissions",
+        headers={"Authorization": f"Bearer {API_TOKEN}"},
+        json=permissions,
+    )
+
+    if not response.ok:
+        print("could not create permissions")
+        response.raise_for_status()
+
+    return response.json()["id"]
+
+
+def create_token(permissions, room_id, task_id=None):
+    response = requests.post(
+        f"{SLURK_API}/tokens",
+        headers={"Authorization": f"Bearer {API_TOKEN}"},
+        json={
+            "permissions_id": permissions,
+            "room_id": room_id,
+            "registrations_left": 1,
+            "task_id": task_id,
+        },
+    )
+
+    if not response.ok:
+        print("could not create token")
+        response.raise_for_status()
+
+    return response.json()["id"]
+
+
+def create_user(name, token):
+    response = requests.post(
+        f"{SLURK_API}/users",
+        headers={"Authorization": f"Bearer {API_TOKEN}"},
+        json={"name": name, "token_id": token},
+    )
+
+    if not response.ok:
+        print(f"could not create user: {name}")
+        response.raise_for_status()
+
+    return response.json()["id"]
+
+
+def find_task_layout_file(path):
+    for filename in path.rglob("*.json"):
+        if all(i in str(filename) for i in ["task", "layout"]):
+            return filename
+
+    print(
```

maybe even raise an error here instead of returning None

---

also below

---

that's better, I changed it so that errors are raised